src/Pate/SimState.hs

```diff
@@ -115,6 +127,159 @@ simOutRegs ::
 simOutRegs simIn = simRegs $ simOutState simIn
 
 
+newtype ExprSet sym tp where
+  ExprSet :: Set (AsOrd (W4.SymExpr sym) tp) -> ExprSet sym tp
+
+data AsOrd f tp where
+  AsOrd :: OrdF f => { unAsOrd :: f tp } -> AsOrd f tp
```

Comment:
I'm surprised this is necessary - interesting pattern (not opposed)

Reply:
It's just used to create a set of a single type of expression. It felt a bit strange, but it was the most obvious way to connect `Ord` and `OrdF` at the type level for `SymExpr`.